sympy/functions/combinatorial/tests/test_comb_factorials.py

```diff
@@ -423,104 +512,155 @@ def test_binomial():
     assert binomial(-10, 1, evaluate=False).is_nonnegative is False
     assert binomial(-10, 7, evaluate=False).is_nonnegative is False
 
-    # issue #14625
-    for _ in (pi, -pi, nt, v, a):
-        assert binomial(_, _) == 1
-        assert binomial(_, _ - 1) == _
-    assert isinstance(binomial(u, u), binomial)
-    assert isinstance(binomial(u, u - 1), binomial)
-    assert isinstance(binomial(x, x), binomial)
-    assert isinstance(binomial(x, x - 1), binomial)
-
-    #issue #18802
-    assert expand_func(binomial(x + 1, x)) == x + 1
-    assert expand_func(binomial(x, x - 1)) == x
-    assert expand_func(binomial(x + 1, x - 1)) == x*(x + 1)/2
-    assert expand_func(binomial(x**2 + 1, x**2)) == x**2 + 1
-
     # issue #13980 and #13981
     assert binomial(-7, -5) == 0
     assert binomial(-23, -12) == 0
     assert binomial(Rational(13, 2), -10) == 0
-    assert binomial(-49, -51) == 0
+    assert binomial(-49, -51) == 1225
 
     assert binomial(19, Rational(-7, 2)) == S(-68719476736)/(911337863661225*pi)
     assert binomial(0, Rational(3, 2)) == S(-2)/(3*pi)
     assert binomial(-3, Rational(-7, 2)) is zoo
     assert binomial(kn, kt) is zoo
 
     assert binomial(nt, kt).func == binomial
-    assert binomial(nt, Rational(15, 6)) == 8*gamma(nt + 1)/(15*sqrt(pi)*gamma(nt - Rational(3, 2)))
-    assert binomial(Rational(20, 3), Rational(-10, 8)) == gamma(Rational(23, 3))/(gamma(Rational(-1, 4))*gamma(Rational(107, 12)))
-    assert binomial(Rational(19, 2), Rational(-7, 2)) == Rational(-1615, 8388608)
-    assert binomial(Rational(-13, 5), Rational(-7, 8)) == gamma(Rational(-8, 5))/(gamma(Rational(-29, 40))*gamma(Rational(1, 8)))
-    assert binomial(Rational(-19, 8), Rational(-13, 5)) == gamma(Rational(-11, 8))/(gamma(Rational(-8, 5))*gamma(Rational(49, 40)))
+    assert unchanged(binomial, nt, Rational(5, 2))
+    assert binomial(kn - 3, kn - 1, evaluate=False).rewrite(gamma) == 0
+    b = binomial(nt, Rational(5, 2))
+    assert b.rewrite(gamma) == b  # nt could be 3/2
```

Comment:
it's okay if nt isn't 3/2 right ? 
`nt` is not an integer and `Rational(5, 2)` is not an integer so the rewrite assumptions hold.

Reply:
see previous regarding cancellation (e.g. Sum) issue